pkg/controller/account/account_finalizer.go

```diff
@@ -3,10 +3,25 @@ package account
 import (
 	"context"
 
+	"github.com/go-logr/logr"
 	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
 	"github.com/openshift/aws-account-operator/pkg/controller/utils"
 )
 
+// Function to add finalizer
+func (r *ReconcileAccount) addFinalizer(reqLogger logr.Logger, account *awsv1alpha1.Account) error {
+	reqLogger.Info("Adding Finalizer for the Account")
+	account.SetFinalizers(append(account.GetFinalizers(), awsv1alpha1.AccountFinalizer))
+
+	// Update CR
+	err := r.Client.Update(context.TODO(), account)
+	if err != nil {
+		reqLogger.Error(err, "Failed to update Account with finalizer")
```

Comment:
Can we actually output the error that happened here? Or probably just wrap the error with this string to add context and return it?